graphql_compiler/compiler/emit_sql.py

```diff
@@ -576,44 +577,34 @@ def _get_fold_outputs(
         # Sort to make select order deterministic.
         return sorted(self._outputs, key=lambda column: column.name, reverse=True)
 
-    # TODO(bojanserafimov): This function communicates both a traversal to a new node and
-    #                       outputting information at the new node. It could be split into two
-    #                       functions that are easier to spec:
-    #                       1. traverse(self, join_descriptor)
-    #                       2. add_output(self, column_name)
-    #                       This way there is no need to define what is considered a folded
-    #                       field. It is up to the caller to expose all they will want to use.
-    def visit_vertex(
-        self,
-        join_descriptor: DirectJoinDescriptor,
-        from_table: Alias,
-        to_table: Alias,
-        current_fold_scope_location: FoldScopeLocation,
-        all_folded_fields: Dict[FoldPath, Set[FoldScopeLocation]],
+    def add_traversal(
+        self, join_descriptor: DirectJoinDescriptor, from_table: Alias, to_table: Alias,
     ) -> None:
-        """Add a new SQLFoldTraversalDescriptor and add outputs, if visiting an output vertex."""
+        """Add a new SQLFoldTraversalDescriptor."""
         if self._ended:
             raise AssertionError(
-                "Cannot visit traversed vertices after end_fold has been called."
+                "Cannot add a traversal after end_fold has been called."
                 f"Invalid state encountered during fold {self}."
             )
 
         self._traversal_descriptors.append(
             SQLFoldTraversalDescriptor(join_descriptor, from_table, to_table)
         )
 
-        # Collect outputs, if there are any at the current FoldScopeLocation.
-        if current_fold_scope_location.fold_path in all_folded_fields:
-            # Ensure that no other outputs have been found for this fold since all outputs must
-            # come from the same FoldScopeLocation.
-            if self._output_vertex_alias is not None:
-                raise AssertionError(
-                    "Cannot visit multiple output vertices in one fold. "
-                    f"Invalid state encountered during fold {self}."
-                )
-            self._output_vertex_alias = to_table
-            self._output_vertex_location = current_fold_scope_location
-            self._outputs = self._get_fold_outputs(current_fold_scope_location, all_folded_fields)
+    def mark_output_location(self, output_table: Alias, output_table_location: FoldScopeLocation):
```

Comment:
```suggestion
    def mark_output_location(self, output_table: Alias, output_table_location: FoldScopeLocation) -> None:
```
Without this, the function is partially-typed and the return type is implicitly `Any`, which doesn't seem to be what we want here.